#! /bin/sh /usr/share/dpatch/dpatch-run
## system_libgsm.dpatch by Tzafrir Cohen <tzafrir.cohen@xorcom.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: use the system copy of libgsm, if availble.

@DPATCH@
diff -urNad asterisk-1.1.9.0beta1.dfsg.1/codecs/codec_gsm.c /tmp/dpep.29zUE5/asterisk-1.1.9.0beta1.dfsg.1/codecs/codec_gsm.c
--- asterisk-1.1.9.0beta1.dfsg.1/codecs/codec_gsm.c	2005-10-12 22:13:37.563900650 +0200
+++ /tmp/dpep.29zUE5/asterisk-1.1.9.0beta1.dfsg.1/codecs/codec_gsm.c	2005-10-12 22:13:50.495196931 +0200
@@ -33,7 +33,7 @@
 #include "asterisk/logger.h"
 #include "asterisk/channel.h"
 
-#include "gsm/inc/gsm.h"
+#include <gsm/gsm.h>
 #include "../formats/msgsm.h"
 
 /* Sample frame data */
diff -urNad asterisk-1.1.9.0beta1.dfsg.1/codecs/Makefile /tmp/dpep.29zUE5/asterisk-1.1.9.0beta1.dfsg.1/codecs/Makefile
--- asterisk-1.1.9.0beta1.dfsg.1/codecs/Makefile	2005-10-12 22:13:37.578898674 +0200
+++ /tmp/dpep.29zUE5/asterisk-1.1.9.0beta1.dfsg.1/codecs/Makefile	2005-10-12 22:18:53.182433117 +0200
@@ -26,10 +26,19 @@
 CFLAGS+=$(shell [ -f $(CROSS_COMPILE_TARGET)/usr/local/include/speex/speex.h ] && echo "-I$(CROSS_COMPILE_TARGET)/usr/local/include/speex")
 CFLAGS+=$(shell [ -f $(CROSS_COMPILE_TARGET)/usr/include/speex/speex.h ] && echo "-I$(CROSS_COMPILE_TARGET)/usr/include/speex")
 
+LIBGSM_PATH:=/usr/local/include /usr/include
+LIBGSM_SYSTEM_HEADERS:=$(wildcard $(LIBGSM_PATH:%=$(CROSS_COMPILE_TARGET)%/gsm/gsm.h))
+ifneq (,$(LIBGSM_SYSTEM_HEADERS))
+LIBGSM=-lgsm
+LIBGSMT=
+else
+LIBGSM=gsm/lib/libgsm.a
+LIBGSMT=$(LIBGSM)
+CFLAGS+=-I.
+endif
+
 LIBG723=g723.1/libg723.a
 LIBG723B=g723.1b/libg723b.a
-LIBGSM=gsm/lib/libgsm.a
-LIBGSMT=gsm/lib/libgsm.a
 LIBLPC10=lpc10/liblpc10.a
 LIBSPEEX=$(shell [ -f $(CROSS_COMPILE_TARGET)/usr/local/lib/libspeex.a ] && echo "-L$(CROSS_COMPILE_TARGET)/usr/local/lib")
 LIBSPEEX+=-lspeex -lm
